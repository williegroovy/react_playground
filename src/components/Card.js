import React from 'react';
import styled from 'styled-components';

const StyledCard = styled.div`
  width: 200px;
  height: 250px;
  margin-bottom: 10px;
  border: 1px solid black;
  border-radius: 5px;
  
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Card = ({ children, face }) => {

  return (
    <StyledCard>
      {[face, children]}
    </StyledCard>
  );
};

export default Card;