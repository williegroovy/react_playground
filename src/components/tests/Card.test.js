import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Card from '../Card';

describe('<Card />', () => {
  test('should be defined', () => {
    expect(Card).toBeDefined();
  });

  test('should render correctly' , () => {
    const wrapper = shallow(
      <Card face={10}>0</Card>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render face={5} correctly' , () => {
    const wrapper = shallow(<Card face={5} />);
    expect(wrapper.contains(5)).toBeTruthy();
  });

  test('should render children correctly' , () => {
    const wrapper = shallow(<Card>16</Card>);
    expect(wrapper.children().text()).toBe("16");
  });

  test('should render children and face correctly' , () => {
    const wrapper = shallow(<Card face={3}>16</Card>);
    console.log('children', wrapper.props().children);
    expect(wrapper.props().children).toEqual([ 3, '16']);
  });
});