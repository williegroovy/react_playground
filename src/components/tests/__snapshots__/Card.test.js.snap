// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    face={10}
  >
    0
  </Card>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        10,
        "0",
      ],
    },
    "ref": null,
    "rendered": Array [
      10,
      "0",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          10,
          "0",
        ],
      },
      "ref": null,
      "rendered": Array [
        10,
        "0",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
